plugin/entryBase.go

```diff
@@ -87,13 +91,42 @@ func (e *EntryBase) setID(id string) {
 	e.washID = id
 }
 
-func (e *EntryBase) getTTLOf(op actionOpCode) time.Duration {
+func (e *EntryBase) getTTLOf(op defaultOpCode) time.Duration {
 	return e.ttl[op]
 }
 
 // OTHER METHODS USED TO FACILITATE PLUGIN DEVELOPMENT
 // AND TESTING
 
+// Name returns the entry's name as it was passed into
+// plugin.NewEntry. You should use e.Name() when making
+// the appropriate API calls within your plugin.
+func (e *EntryBase) Name() string {
+	return e.name()
+}
+
+// SetInitialAttributes sets the entry's initial attributes. Use it
+// after creating the entry via a call to NewEntry.
+func (e *EntryBase) SetInitialAttributes(attr EntryAttributes) {
+	e.attr = attr
+}
+
+// Sync syncs the attribute with the given key's value in the entry's
+// metadata whenever the latter's refreshed via a call to plugin.CachedMetadata.
+// Use Sync if you expect the attribute to change.
+//
+// TODO: Find a way to let plugin authors specify a munging function. We'll
+// need this for the State attribute, and in case the metadata returns some of
+// the other attributes in a different format (e.g. like an arbitrarily formatted
+// string for one of the time fields)
+func (e *EntryBase) Sync(attr SyncableAttribute, key string) {
```

Comment:
Was thinking of making this something like
```
Sync(SyncableAttribute, key, mungeFunc...)
```

(using Go's variadic args to simulate optional input).

Reply:
Or just another function like `SyncMunged`.